src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/ILogger.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// A interface for logging.  It is supported by System.Diagnostics.Tracing.Logger, but it could in theory have
+    /// other implementations.   See Logger for more details on expected use.  
+    /// </summary>
+    public interface ILogger
+    {
+        /// <summary>
+        /// returns true if the given verbosity level is active (will be logged if 'Log' is called).  
+        /// </summary>
+        bool IsEnabled(LogLevel level);
+
+        /// <summary>
+        /// Log a single message.   A message has a name, a Verbosity level and a object that contains any other
+        /// information needed by the message.   Typical this object is created using an anonymous type.  
```

Comment:
Also, I'm not deeply familiar with the language but believe F# may lack anonymous types.